src/main/java/org/folio/rest/impl/EholdingsConfigurationImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import org.folio.config.RMAPIConfigurationClient;
+import org.folio.http.ConfigurationClientProvider;
+import org.folio.rest.converter.RMAPIConfigurationConverter;
+import org.folio.rest.jaxrs.model.EholdingsConfigurationPutApplicationVndApiJson;
+import org.folio.rest.jaxrs.model.EholdingsConfigurationPutApplicationVndApiJsonImpl;
+import org.folio.rest.jaxrs.resource.EholdingsConfiguration;
+import org.folio.rest.util.HeaderConstants;
+import org.folio.rest.validator.HeaderValidator;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+public class EholdingsConfigurationImpl implements EholdingsConfiguration {
+
+  private RMAPIConfigurationClient configurationClient;
+  private RMAPIConfigurationConverter converter;
+  private HeaderValidator headerValidator;
+
+  public EholdingsConfigurationImpl() {
+    this(new RMAPIConfigurationClient(new ConfigurationClientProvider()), new RMAPIConfigurationConverter(), new HeaderValidator());
+  }
+
+  public EholdingsConfigurationImpl(RMAPIConfigurationClient configurationClient, RMAPIConfigurationConverter converter, HeaderValidator headerValidator) {
+    this.configurationClient = configurationClient;
+    this.converter = converter;
+    this.headerValidator = headerValidator;
+  }
+
+  @Override
+  public void getEholdingsConfiguration(Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    if (!headerValidator.validate(okapiHeaders, asyncResultHandler)) {
+      return;
+    }
+    configurationClient.retrieveConfiguration(
+      okapiHeaders.get(HeaderConstants.OKAPI_TOKEN_HEADER),
+      okapiHeaders.get(HeaderConstants.OKAPI_TENANT_HEADER),
+      okapiHeaders.get(HeaderConstants.OKAPI_URL_HEADER))
+      .thenAccept(rmAPIconfiguration -> {
+        EholdingsConfigurationPutApplicationVndApiJsonImpl configuration = converter.convert(rmAPIconfiguration);
+        asyncResultHandler.handle(Future.succeededFuture(GetEholdingsConfigurationResponse.respond200WithApplicationVndApiJson(configuration)));
+      })
+      .exceptionally(throwable -> {
```

Comment:
Not sure about this, if this will end up throwing a stack trace, instead can we throw an InternalServerError?

Reply:
This returns code 500 with body "Server Error"
now that I looked into it closer I changed it to always log exception before returning error.

Reply:
By throwing InternalServerError do you mean returning 500 or something else?

Reply:
I was thinking something on these lines: 
`asyncResultHandler.handle(Future.succeededFuture(Response.respond500WithTextPlain(throwable.getCause().getMessage())));` - where the Response class is whatever class we are using there which in this case is `GetEholdingsConfigurationResponse`

Reply:
Sure, will add this to raml:
```      500:
        description: "Internal server error"
        body:
          text/plain:
            example: "Internal server error, contact administrator"
```

also this is a case that catches all exceptions, so throwable.getCause().getMessage() might not have a high level message, if it is not ok, then we can use "Internal server error" for all exceptions and log the message with stacktrace